Gravity/Gravity/Extensions/RdoExtensions.cs

```diff
@@ -54,65 +56,62 @@ public static T ToHydratedDto<T>(this RDO rdo)
 							newValueObject = theFieldValue.ValueAsLongText;
 							break;
 						case (int)RdoFieldType.MultipleChoice:
-							// Means we have IList<some_enum> here, in fieldAttribute.ObjectDTOType
-							var valueAsMultipleChoice = theFieldValue.ValueAsMultipleChoice;
-							if (valueAsMultipleChoice != null)
 							{
-								var listOfEnumValuesDoNotUse = typeof(List<>).MakeGenericType(fieldAttribute.ObjectFieldDTOType);
-								var listOfEnumValuesInstance = (IList)Activator.CreateInstance(listOfEnumValuesDoNotUse);
-								foreach (var choice in valueAsMultipleChoice)
+								var valueAsMultipleChoice = theFieldValue.ValueAsMultipleChoice;
+								if (valueAsMultipleChoice == null)
+									break;
+
+								//get a List<target_enum_type> to hold your converted values
+								var genericListType = typeof(List<>).MakeGenericType(fieldAttribute.ObjectFieldDTOType);
+								var listOfEnumValuesInstance = (IList)Activator.CreateInstance(genericListType);
+
+								//get choice names
+								var choiceNames = new HashSet<string>(
+									valueAsMultipleChoice.Select(c => c.Name.ChoiceTrim()),
+									StringComparer.InvariantCultureIgnoreCase);
+
+								//get enum values of type that correspond to those names
+								var enumValues = Enum.GetValues(fieldAttribute.ObjectFieldDTOType).Cast<Enum>()
+									.Where(x => choiceNames.Contains(x.ToString()));
+
+								//add to list
+								foreach (var theValueObject in enumValues)
 								{
-									string choiceNameTrimmed = choice.Name.Replace(" ", "").Replace("-", "").Replace("(", "").Replace(")", "");
-									System.Array enumValues = System.Enum.GetValues(fieldAttribute.ObjectFieldDTOType);
-									for (int i = 0; i < enumValues.Length; i++)
-									{
-										object theValueObject = enumValues.GetValue(i);
-										if (theValueObject.ToString().Equals(choiceNameTrimmed, StringComparison.OrdinalIgnoreCase) == true)
-										{
-											listOfEnumValuesInstance.Add(theValueObject);
-										}
-									}
+									listOfEnumValuesInstance.Add(theValueObject);
 								}
 
-								// Now we have a List<object> and we need it to be List<fieldAttribute.ObjectFieldDTOType>
+								//set to new object
 								newValueObject = listOfEnumValuesInstance;
 							}
 							break;
 						case (int)RdoFieldType.MultipleObject:
-							newValueObject = theFieldValue.GetValueAsMultipleObject<kCura.Relativity.Client.DTOs.Artifact>()
-								.Select<kCura.Relativity.Client.DTOs.Artifact, int>(artifact => artifact.ArtifactID).ToList();
+							newValueObject = theFieldValue.GetValueAsMultipleObject<Artifact>()
+								.Select(artifact => artifact.ArtifactID).ToList();
 							break;
 						case (int)RdoFieldType.SingleChoice:
-							kCura.Relativity.Client.DTOs.Choice theChoice = theFieldValue.ValueAsSingleChoice;
-							if (theChoice != null)
 							{
```

Comment:
You missed to remove some "{}" brackets on lines 92 & 102.

Reply:
All is good, my bad. Afterwards I saw that you have left "{}" brackets for each of the cases in the switch. Which makes sense, as it is more readable that way.